docs/rfc/rfc-19-random-beacon-v2.adoc

```diff
@@ -0,0 +1,252 @@
+which relay entry is submitted with a gas limit of 2M. This allows applications
+using the random beacon to avoid an additional complexity at their end but comes
+at the cost of random beacon operators who need to have enough ETH on their
+balance. This approach is not compatible with the idea that all rewards will be
+solely in T given that it is impossible to establish the cost of executing a
+callback in T.
+
+The fact the full - even the most complex - callback is executed in the same
+transaction in which the relay entry is submitted gives an impression of better
+security. This impression is false though, given that the entry to be submitted
+is visible in the mempool and smart attackers can have their transactions mined
+faster to, for example, put the sortition pool using the entry for group
+selection in the desired state. This issue needs to be solved on the sortition
+pool side with initiation time for new operators in the pool and/or state lock.
+
+Instead of executing any callback in the same transaction in which relay entry
+is submitted, we should allow only simple, low gas budget callbacks with a gas
+limit controlled by the governance. Applications wanting to use a relay entry
+should submit another transaction using the relay entry value previously set by
+the random beacon. They should also employ an additional security check making
+sure the entry submitted is only valid for a certain number of blocks to avoid
+the situation when relay entry beacon transaction and application-specific
+transaction are executed far from each other.
+
+Smart contract consuming new relay entry needs to implement `IRandomBeaconConsumer`
+interface. Gas limit for `__beaconCallback` call should be initially set to 50k
+gas which is enough to SSTORE new relay entry, block height in which the entry
+was submitted, and to emit an event. Callback gas limit should be a governable
+value. Failure in the callback function should not revert the relay entry
+transaction. When requesting a relay entry, it should be possible to pass an
+optional address parameter - this is the address of a contract implementing
+`IRandomBeaconConsumer` interface that should be called when a new relay entry
+is submitted to the chain.
+
+```
+interface IRandomBeaconConsumer {
+    /// @notice Receives relay entry produced by Keep Random Beacon. This function
+    /// should be called only by Keep Random Beacon.
+    ///
+    /// @param relayEntry Relay entry (random number) produced by Keep Random
+    ///                   Beacon.
+    /// @param blockNumber Block number at which the relay entry was submitted
+    ///                    to the chain.
+    function __beaconCallback(uint256 relayEntry, uint256 blockNumber) external;
+}
+```
+
+=== Timeouts
+
+We should extend the timeout for submitting relay entry to give operators more
+time to react. After 64x10 blocks from the relay request, if no entry was
```

Comment:
It is not clear to me whether the `64x10 blocks` number comes from `group size` x `the number of blocks for a member to become eligible to submit relay entry` or is it a hardcoded or governable value.